### ISSUE TITLE
AnnotationUseStyle: compact_no_array does not violate extended and multi param annotations

### ISSUE THREAD

**Issue by eric-milles** [checkstyle/checkstyle#3210]
```
/var/tmp $ cat TestClass.java 
@SuppressWarnings(value = "rawtypes") // 'value=' is superflous (non-compact) - VIOLATION
public class Example {
    @SuppressWarnings({"unchecked"}) // array is superflous - VIOLATION 
    public void somePublicMethod() {
    }

    @SomeAnnotation(attr1="value1", arrt2={"value2"}) // array is superfluous  - VIOLATION
    public Object[] stuff;
}

/var/tmp $ cat my_check.xml 
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">
<module name = "Checker">
    <module name="TreeWalker">
        <module name="AnnotationUseStyle">
            <property name="elementStyle" value="compact_no_array" />
            <property name="trailingArrayComma" value="ignore" />
        </module>
    </module>
</module>

/var/tmp $ java -jar checkstyle-7.0-all.jar -c my_check.xml TestClass.java 
Starting audit...
[ERROR] /var/tmp/TestClass.java:3: Annotation style must be 'COMPACT_NO_ARRAY'. [AnnotationUseStyle]
Audit done.
Checkstyle ends with 1 errors.
```

Expected violations are marked at source code - there should be 3 of them as all of them are not compact.

Attention: "compact" mode should give 1 violation 1st line.

http://checkstyle.sourceforge.net/config_annotation.html#AnnotationUseStyle
http://checkstyle.sourceforge.net/property_types.html#elementStyle
##

**Comment by romani** [comment]
@eric-milles , please change description of issue to be exact and certain for problem.

**Comment by eric-milles** [comment]
@SomeAnnotation(attr1="value1", arrt2={"value2"}) // array is superfluous

**Comment by romani** [comment]
Looks like you try to point to bug:

```
/var/tmp $ cat TestClass.java 
@SuppressWarnings(value = "rawtypes") // 'value=' is superflous but ok for our config
public class Example {
    @SuppressWarnings({"unchecked"}) // array is superflous - VIOLATION 
    public void somePublicMethod() {
    }

    @SomeAnnotation(attr1="value1", arrt2={"value2"}) // array is superfluous  - VIOLATION
    public Object[] stuff;
}

/var/tmp $ cat my_check.xml 
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">
<module name = "Checker">
    <module name="TreeWalker">
        <module name="AnnotationUseStyle">
            <property name="elementStyle" value="compact_no_array" />
            <property name="trailingArrayComma" value="ignore" />
        </module>
    </module>
</module>

/var/tmp $ java -jar checkstyle-7.0-all.jar -c my_check.xml TestClass.java 
Starting audit...
[ERROR] /var/tmp/TestClass.java:3: Annotation style must be 'COMPACT_NO_ARRAY'. [AnnotationUseStyle]
Audit done.
Checkstyle ends with 1 errors.
```

Expected behaviour is violation at line 3 and line 7. Am I reading your mind correctly ?

**Comment by eric-milles** [comment]
I'm sorry for the confusion caused by my initial description and follow-up comment.  My overall goal is to have `<module name="AnnotationUseStyle">
            <property name="elementStyle" value="compact_no_array" />
            <property name="trailingArrayComma" value="ignore" />
        </module>` produce a violation for each of the cases where extra constructs are present.  Some of them are working correctly, like `@SuppressWarnings({"unchecked"})` should be a violation.  And some are not, like `@SomeAnnotation(attr1="value1", arrt2={"value2"})`.

In your example, all three annotations should produce a violation.

**Comment by izishared** [comment]
I'm on it.

**Comment by Sekair** [comment]
I am on it

**Comment by Sekair** [comment]
@eric-milles @romani There is some problem with this issue.
`@SomeAnnotation(attr1="value1", arrt2={"value2"}) // array is superfluous  - VIOLATION`
This must be a valid example with no violation, cuz the documentaiton says

> compact_no_array | It is similar to the compact style but single value arrays are flagged. With annotations a single value array does not need to be placed in an array initializer. **This style can only be used when there is an element called 'value' which is either the sole element or all other elements have default values.**

What should i do? I dont know why we need to have "value" element to validate a single value array, so we can allow to mark a single value array in any element as violate and just use "value" to validate "compact" rule. Or we can just ignore it?

**Comment by Vladlis** [comment]
+1 from me for
> mark a single value array in any element

**Comment by romani** [comment]
@Sekair , we need to validate all elements of annotation not only default("value").
Documentation at http://checkstyle.sourceforge.net/property_types.html#elementStyle is very uncertain, please help us to fix it. Each value should have description that explain behavior , with examples where violation will be and where not.
Please update section - https://github.com/checkstyle/checkstyle/blob/master/src/xdocs/property_types.xml#L512 to avoid any future misinterpretations of how works.
Doc polishing could be done in separate commit of the same PR or in the same commit.

I appreciate your help, thanks in advance.

**Comment by romani** [comment]
fix is merged.
